dict/dict0dict.cc

```diff
@@ -621,14 +611,13 @@ bool dict_table_rename_in_cache(dict_table_t *table, const char *new_name, bool
   old_size = mem_heap_get_size(table->heap);
   old_name = table->name;
 
-  fold = ut_fold_string(new_name);
-
   /* Look for a table with the same name: error if such exists */
   {
-    dict_table_t *table2;
-    HASH_SEARCH(
-      name_hash, dict_sys->table_hash, fold, dict_table_t *, table2, ut_ad(table2->cached), (strcmp(table2->name, new_name) == 0)
-    );
+    dict_table_t *table2{nullptr};
+    if (auto itr = dict_sys->table_hash->find(new_name); itr != dict_sys->table_hash->end()) {
```

Comment:
We should use "it" for all iterators, so that it's consistent.

Reply:
Done